lib/typescript/render/components/RichCardCarousel/index.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {useState} from 'react';
+import styles from './index.module.scss';
+import {MediaRenderProps} from '../RichCard/Media';
+import {DefaultMessageRenderingProps} from '../index';
+import {RichCard} from '../RichCard';
+import leftArrow from 'assets/images/icons/leftArrow.svg';
+import rightArrow from 'assets/images/icons/rightArrow.svg';
+
+type Suggestions = [
+  {
+    reply: {
+      text: string;
+      postbackData: string;
+    };
+  },
+  {
+    reply?: {
+      text: string;
+      postbackData: string;
+    };
+  }
+];
+
+type Card = {
+  id?: string;
+  title?: string;
+  description?: string;
+  media: MediaRenderProps;
+  suggestions: Suggestions;
+};
+
+enum Direction {
+  back = 'back',
+  next = 'next',
+}
+
+enum Width {
+  short = 'SHORT',
+  medium = 'MEDIUM',
+}
+
+// Given width of RichCards
+enum CardWidth {
+  short = 136,
+  medium = 280,
+}
+
+enum VisibleArea {
+  short = 176,
+  medium = 320,
+}
+
+export type RichCardCarouselRenderProps = DefaultMessageRenderingProps & {
+  cardWidth: string;
+  cardContents: [Card];
+  id: string;
+  inChatPlugin: boolean;
+};
+
+export const RichCardCarousel = (props: RichCardCarouselRenderProps) => {
+  const {cardContents, cardWidth, id, fromContact, inChatPlugin} = props;
+  const [position, setPosition] = useState(0);
+  const [disabled, setDiabled] = useState(false);
+  const amountCards = cardContents.length;
+
+  const button = (position: number, amountCards: number, id: string) => {
+    if (position == 0) {
+      return (
+        <button
+          className={styles.moveNext}
```

Comment:
if a button is not a submit in a form, you should always add a `type='button'` to it.  The default type is submit sadly.

Reply:
Same with the other buttons below